### PR TITLE
253 - Fix do not show pagination if there is only 1 page + files results copy

### LINKED ISSUE #253: Pager should not be shown when there are not enough items + wrong message for files count
## Overview of the Feature Request

Pager should not be shown when there are not enough items.

On the other hand, the SPA should replicate JSF for the files count title: "1 File" instead of "1 to 1 of 1 Files". We should double check how the copy looks like for more file counts to make sure the behavior is the same.

I encountered this while testing: https://github.com/IQSS/dataverse-frontend/pull/227

SPA:
spatab

JSF:
jsftab

## What kind of user is the feature intended for?
All of them

## What inspired the request?
- https://github.com/IQSS/dataverse-frontend/pull/227

## What existing behavior do you want changed?
Pager should not be shown when there are not enough items

## Any brand new behavior do you want to add to Dataverse?
None

## Any open or closed issues related to this feature request?
- https://github.com/IQSS/dataverse-frontend/pull/227

### PR DESCRIPTION
**Pull request by MellyGray**
## What this PR does / why we need it:
This PR fixes 2 bugs in the Files Table:

1 - Pagination controls shouldn't be shown if the total number of files would result in just  1 page. The expected behaviour is showing the pagination controls when there is more than 1 page of files.
2 - When there is only 1 File in the table the results copy should be ✅  '1 File' instead of ❌ '1 to 1 of 1 Files'

## Which issue(s) this PR closes:

- Closes #253 

### Step 1: Run the development environment
1. Run `npm I`
3. `cd packages/design-system && npm run build`
4. `cd ../../`
5. Check that you have a  `.env` file such as the .env.example, with the `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000` variable
6.  `cd dev-env`
7. `./run-env.sh unstable`
8. To check the environment go to <http://localhost:8000> and check your local dataverse installation

### Step 2: Test Dataset View mode with the implemented changes to download the files
1. Go to <http://localhost:8000>
2. Login as admin using username: dataverseAdmin and password: admin1
3. Create a new Dataset
4. Upload some files to the dataset
5. From the dataset view mode copy the search parameters from the url. Ex.: `?persistentId=doi:10.5072/FK2/LHGRHP&version=DRAFT`
6. Go to <http://localhost:8000/spa/datasets> and paste the search parameters. Ex.: <http://localhost:8000/spa/datasets?persistentId=doi:10.5072/FK2/LHGRHP&version=DRAFT>
8. You are currently viewing the same dataset in the SPA. In the files table, check the copy and the pagination bugs. Upload or remove files as necessary to test the bug.

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
No

## Is there a release notes update needed for this change?:
No

### CODE REVIEW

File: tests/component/sections/dataset/dataset-files/DatasetFiles.spec.tsx [hunk 1]
```diff
@@ -116,7 +116,7 @@ describe('DatasetFiles', () => {
       cy.findByRole('columnheader', { name: '101 to 150 of 200 Files' }).should('exist')
     })
 
-    it('renders the first page if there is only one page and the user changes to a lower page size', () => {
+    it('does not render pagination if the user changes to a lower page size resulting in one page', () => {
```
Comment (M27Mangan) [inline tests/component/sections/dataset/dataset-files/DatasetFiles.spec.tsx:119]: Shouold this be "... if the user changes to a **_larger_** page size..." ?

**Review by M27Mangan — CHANGES_REQUESTED** [review]
I would like to also explore hiding the buttons for "First" and "Last" if there are only two pages.

**Comment by MellyGray** [comment]
> I would like to also explore hiding the buttons for "First" and "Last" if there are only two pages.

@M27Mangan Why would you want to hide those? They exist in the JSF frontend. Furthermore, it could be confusing for users to see the buttons change just for two pages. While the "First" and "Last" buttons may be redundant with only two pages, I believe it's better to maintain this redundancy rather than introduce too many exceptions to the User Interface, which can be confusing for users

image

File: tests/component/sections/dataset/dataset-files/DatasetFiles.spec.tsx (hunk 1 above)
Comment (MellyGray) [reply tests/component/sections/dataset/dataset-files/DatasetFiles.spec.tsx]: Thanks, I fixed the text

**Comment by M27Mangan** [comment]
Review complete, moving to Ready for QA!

**Comment by ekraffmiller** [comment]
looks good! Merging

**Pull request merged**